File: test/cs-stub/BaboonTests/MapKeyMalformedTests.cs
// PR-F (M24) — cross-language malformed map-key error consistency.
//
// Verifies that decoding a JSON object whose map-key cannot be parsed back into the
// id type throws BaboonCodecException.DecoderFailure with message containing
// "malformed key". Replaces the prior unchecked cast that produced InvalidCastException.
//
// Uses the my.ok.m19.singleid fixture; generated symbols are produced by
// mdl :test-gen-regular-adt under target/test-regular/cs-stub/.
#nullable enable

using Baboon.Runtime.Shared;
using Newtonsoft.Json.Linq;
using NUnit.Framework;

namespace ConversionsTest
{
    [TestFixture]
    public class MapKeyMalformedTests
    {
        private readonly BaboonCodecContext _ctx = BaboonCodecContext.Compact;

        [Test]
        public void HolderJsonDecodeThrowsDecoderFailureForMalformedMapKey()
        {
            var badJson = "{\"m\":{\"not_a_valid_id\":\"v\"}}";
            var node = JToken.Parse(badJson);
            var ex = Assert.Throws<BaboonCodecException.DecoderFailure>(() =>
                My.Ok.M19.Singleid.Holder_JsonCodec.Instance.Decode(_ctx, node));
            Assert.That(ex!.Message, Does.Contain("malformed key"));
        }
    }
}

File: test/sw-stub/Tests/RuntimeTests/MapKeyMalformedTests.swift
// PR-F (M24) — cross-language malformed map-key error consistency.
//
// Verifies that decoding a JSON object whose map-key cannot be parsed back into the
// id type throws BaboonCodecException.decoderFailure with message containing
// "malformed key". Replaces the prior fatalError-on-Left and forced cast.
//
// Uses the my.ok.m19.singleid fixture; generated symbols are produced by
// mdl :test-gen-regular-adt under target/test-regular/sw-stub/.

import XCTest
import Foundation
@testable import BaboonRuntime
@testable import MyOkM19Singleid

final class MapKeyMalformedTests: XCTestCase {

    private let ctx = BaboonCodecContext.compact

    func testHolderJsonDecodeThrowsDecoderFailureForMalformedMapKey() {
        let badJson: [String: Any] = ["m": ["not_a_valid_id": "v"]]
        do {
            _ = try Holder_JsonCodec.instance.decode(ctx, badJson)
            XCTFail("expected throw, got success")
        } catch let err as BaboonCodecException {
            XCTAssertTrue(err.description.contains("malformed key"),
                "expected 'malformed key' in error description; got: \(err.description)")
        } catch {
            XCTFail("expected BaboonCodecException; got \(type(of: error))")
        }
    }
}
